mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulProfileScreen.kt

```diff
@@ -36,8 +34,14 @@ fun StatefulProfileScreen(
     modifier: Modifier = Modifier,
 ) {
   val socialFacade = LocalSocialFacade.current
+  val chessFacade = LocalChessFacade.current
   val profile by remember(socialFacade, uid) { socialFacade.profile(uid) }.collectAsState(null)
-  val state = profile?.let { FetchedUserProfileScreenState(it) }
+  val matches =
+      remember(chessFacade, profile) { profile?.let { chessMatches(it, chessFacade) } }
+          ?.collectAsState(emptyList())
+          ?.value
+          ?: emptyList()
+  val state = profile?.let { FetchedUserProfileScreenState(it, matches) }
```

Comment:
I think you should use the same pattern as here (https://github.com/epfl-SDP/android/blob/main/mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt#L91-L151) to manage the state using the facades and an eventual `CoroutineScope`.